### DISCUSSION TITLE
how configure panes layout when clicking new tab

### DISCUSSION THREAD

**Discussion by nbenitez** [wezterm/wezterm#3364] [Q&A] [answered]
Hi all, I configured wezterm so when it starts it creates a 2X2 pane layout (my favourite layout for a big monitor) I do it like so:
```
wezterm.on('gui-startup', function(cmd)
  local tab, pane, window = mux.spawn_window(cmd or {})
  -- maximize window
  window:gui_window():maximize()
  -- Create a 2x2 pane configuration
  local tpane = pane:split { direction = 'Top', size = 0.5 }
  pane:split { direction = 'Right', size = 0.5 }
  tpane:split { direction = 'Right', size = 0.5 }
end)
```

However, then when clicking "+" for new tab I would like a similar 2X2 layout be created, but instead a single pane is created and shown, so:

how can I do it? what event is triggered by clicking "new tab" that I can connect to and run my pane configuration?

Thank you,

**Comment by wez** [comment, marked as answer]
What you need is:
* https://github.com/wez/wezterm/issues/323

I just pushed an implementation of that, and it should be available in the nightly build in ~1 hour.

Docs already available at: https://wezfurlong.org/wezterm/config/lua/window-events/new-tab-button-click.html

**Reply by nbenitez** [reply]
You're awesome 👍🏻   I will try that.
